include/tvm/tir/builtin.h

```diff
@@ -626,6 +626,13 @@ TVM_DLL const Op& texture2d_store();
  */
 TVM_DLL const Op& texture2d_load();
 
+/*!
+ * \brief Copy 1d memory from source to destination
+ * Same semantics as memcpy(destination, source, size)
+ * Allows for device specific implementations e.g. direct memory access (DMA)
+ */
+TVM_DLL const Op& mem_copy();
```

Comment:
cc @tqchen @junrushao1994 if they want a better name here.

Reply:
Is it possible to make it more specific? We have a few examples above like texture2d_load/store. What about dma_mem_copy() or any hexagon-specific name?

Reply:
I was thinking about something with "DMA" in the name but I realized that DMA is more of an implementation detail.  I ended up referencing DMA in the comments instead.  Happy to take suggestions including `dma_mem_copy`.

Reply:
I thought about:
* `mem_copy_local` (to emphasize the local nature of the intended usage, unlike `TVMArrayCopyFromTo` @junrushao1994 mentioned which might involve RPC
* `mem_copy_1d` (if we intend to add 2d rectangular memcpy later)

But both of them don't particularly sound better.